### PR TITLE
Multilevel ibfe 3

### PR DESCRIPTION
**Pull request by drwells**
This PR sets up the input database that lets users specify the subdomain-to-level mapping.

The new code checks that the user doesn't specify the level more than once - I want to explicitly check that this error works so I also extended attest to work with tests that are expected to fail at runtime.

### IBAMR Pull Request Checklist
- [x] Does the test suite pass?
- [x] Was clang-format run?
- [x] Were relevant issues cited? Please remember to add `Fixes #12345` to close
      the issue automatically if we are fixing the problem.
- [x] Is this a change others will want to know about? If so, then has a
      changelog entry been added?
- [x] If new data structures have been added to a class then have they been set
      up appropriately for restarts? If so, ensure that the restart version
      number is incremented.
- [x] Does this change include a bug fix or new functionality? If so, a new test
      or tests should be added. New tests should run quickly (less than a minute
      in release mode). If possible, an older test should gain a new option so
      that we do not need to compile more test executables.
- [x] Did you (if your account has permission to do so) set relevant labels on
      GitHub for the pull request?

### CODE REVIEW

File: ibtk/include/ibtk/FEDataManager.h [hunk 1]
```diff
@@ -284,6 +334,23 @@ class FEData : public SAMRAI::tbox::Serializable
...
+ *   level_3 = 10, 12, 14
+ *   level_5 = 1000, 1003, 1006, 1009
+ * }
+ * @endcode
+ * This particular input will associate all elements with subdomain id 4 with
+ * patch level 1, all elements with subdomain ids 10, 12, or 14 with patch level
+ * 3, etc. All unspecified subdomain ids will be associated with the finest
+ * patch level. All inputs in this database for levels finer than the finest
+ * level are ignored (e.g., if the maximum patch level number is 4, then the
+ * values given in the example for level 5 ultimately end up on level 4).
```
Comment (drwells) [inline ibtk/include/ibtk/FEDataManager.h:352]: The documentation is ahead of what is actually implemented.

File: ibtk/src/lagrangian/FEDataManager.cpp [hunk 2]
```diff
@@ -632,9 +667,9 @@ FEDataManager::reinitElementMappings()
                 "At least one node in the current mesh is inside the fluid domain and not associated with any "
                 "patch. This class currently assumes that all elements are on the finest level and will not "
                 "work correctly if this assumption does not hold. This usually happens when you use multiple "
-                "patch levels and set the regrid CFL interval to a value larger than one. To change this check set "
-                "node_outside_patch_check to a different value in the input database: see the documentation of "
-                "FEDataManager for more information.";
+                "patch levels and set the regrid CFL interval to a value larger than one. To change this check "
+                "set node_outside_patch_check to a different value in the input database: see the documentation "
+                "of FEDataManager for more information.";
```
Comment (drwells) [inline ibtk/src/lagrangian/FEDataManager.cpp:672]: I'm not sure why clang-format started complaining about this.

File: ibtk/include/ibtk/FEDataManager.h [hunk 3]
```diff
@@ -1078,6 +1145,16 @@ class FEDataManager : public SAMRAI::tbox::Serializable
     void collectActivePatchNodes(std::vector<std::vector<libMesh::Node*> >& active_patch_nodes,
                                  const std::vector<std::vector<libMesh::Elem*> >& active_patch_elems);
 
+    /*!
+     * Store the association between subdomain ids and patch levels.
+     */
+    SubdomainToLevelTranslation d_level_lookup;
```
Comment (boyceg) [inline ibtk/include/ibtk/FEDataManager.h:1151]: FWIW, although this is not totally consistent, I think we usually do not expose member variables directly in the public interface.

Comment (boyceg) [reply ibtk/include/ibtk/FEDataManager.h:1151]: Oh, duh, this is not in the public interface.

**Review by boyceg — APPROVED** [review]

**Comment by boyceg** [comment]
Looks good!

File: ibtk/src/lagrangian/FEDataManager.cpp (hunk 2 above)
Comment (boyceg) [reply ibtk/src/lagrangian/FEDataManager.cpp:672]: 🤷‍♂️

File: ibtk/include/ibtk/FEDataManager.h (hunk 1 above)
Comment (boyceg) [reply ibtk/include/ibtk/FEDataManager.h:352]: 👍

**Pull request merged**